src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -162,7 +159,7 @@ Error DatasetManager::HandleSet(Coap::Message &aMessage, const Ip6::MessageInfo
     }
 
     // check active timestamp rollback
-    if (IsPendingDataset() && (!hasNetworkKey || !doesAffectNetworkKey))
+    if (IsPendingDataset() && !doesAffectNetworkKey)
```

Comment:
Adding a quick note on this: We don't need the `!hasNetworkKey` check as when `hasNetworkKey` is `false`, then `doesAffectNetworkKey` will also be `false`.